021-presence-service.md

```diff
@@ -0,0 +1,163 @@
+
+- [RFC Template](#rfc-template)
+  - [Summary](#summary)
+  - [Resources](#resources)
+  - [Implementation](#implementation)
+    - [API Changes](#api-changes)
+    - [Workers / Commands](#workers--commands)
+    - [Supporting Libraries](#supporting-libraries)
+    - [Data Structures](#data-structures)
+    - [SDKs](#sdks)
+    - [Breaking Changes](#breaking-changes)
+    - [Documentation & Content](#documentation--content)
+  - [Reliability](#reliability)
+    - [Security](#security)
+    - [Scaling](#scaling)
+    - [Benchmarks](#benchmarks)
+    - [Tests (UI, Unit, E2E)](#tests-ui-unit-e2e)
+  - [Open Questions](#open-questions)
+  - [Future Possibilities](#future-possibilities)
+
+## Summary
+<!-- Describe the problem we want to solve and suggest a solution in a few paragraphs -->
+
+In many use cases where we build an application that requires realtime collaboration, we need to know which of our users are online and what their status is. Today, this is possible in Appwrite only by building a complex custom integration. This can be relevant to collaborative apps like Slack, Google Docs, or multiplayer gaming apps. 
+
+To reduce friction, make this a breeze, and enable new use cases for using Appwrite, we want to introduce a new presence service that will also be integrated with the Appwrite realtime server REST and GraphQL APIs and will allow each user to set strict privacy permissions for who can see their current presence and what is their current status.
+
+## Resources
+<!-- List all the resources used for writing this RFC -->
+
+## Implementation
+
+<!-- Write an overview to explain the suggested implementation -->
+
+### API Changes
+<!-- Do we need new API endpoints? List and describe them and their API signatures -->
+
+```POST /v1/presences```
+
+Create a presence. Set the user presence status, expiry (minutes/date - TBD), and permissions for who has access to see the current presence. This will imitate the equivalent action in the GraphQL and Realtime APIs.
+
+```GET /v1/presences```
+
+List all presences, on client side this will show only presneces I have permission to view. On server it will show all presences. This endpoint will support queries and pagination like any other list endpoint on Appwrite.
+
+```GET /v1/presences/:id```
+
+Get presence by ID.
+
+```PUT /v1/presences/:id```
+
+Update presnece by ID.
+
+```DELETE /v1/presences/:id```
+
+Delete presence by ID.
+
+### Events
+
+The new endpoints will trigger standard events that will be used for listening to Realtime API changes or to execute functions and webhooks.
+
+### Console updates
+
+- We should implement presence in the console itself and make current org members online or not or last online
+- We will add a new section under auth called `Presences`, this section will show a dashboard with usage showing how many people were online in a specific time range. Below that, we will show avatars of all currently online users or users who just recently went offline. 
```

Comment:
Hmm I imagined this as tab under user detail page. Can 1 user have multiple presences?